src/corespotlight.cs

```diff
@@ -979,6 +989,38 @@ interface CSSearchableItemAttributeSet : NSCopying, NSSecureCoding {
 		[Export ("valueForCustomKey:")]
 		[return: NullAllowed]
 		INSSecureCoding ValueForCustomKey (CSCustomAttributeKey key);
+
+		// CSSearchableItemAttributeSet_CSGeneral
+
+		[iOS (11,0), NoTV, Mac (10, 11)]
+		[NullAllowed, Export ("userCreated", ArgumentSemantic.Strong)]
+		NSNumber IsUserCreated { [Bind ("isUserCreated")] get; set; }
+
+		[iOS (11, 0), NoTV, Mac (10, 11)]
+		[NullAllowed, Export ("userOwned", ArgumentSemantic.Strong)]
+		NSNumber IsUserOwned { [Bind ("isUserOwned")] get; set; }
+
+		[iOS (11, 0), NoTV, Mac (10, 11)]
+		[NullAllowed, Export ("userCurated", ArgumentSemantic.Strong)]
+		NSNumber IsUserCurated { [Bind ("isUserCurated")] get; set; }
+
+		[iOS (11, 0), NoTV, Mac (10, 11)]
+		[NullAllowed, Export ("rankingHint", ArgumentSemantic.Strong)]
+		NSNumber RankingHint { get; set; }
```

Comment:
By looking at the headers:

+//This property allows content donors to provide a ranking signal to each indexed item
+//It will inform the ranker, allowing it to distinguish more easily between otherwise similar items
+//This is query-independent and should be used to indicate the relative importance of an item w.r.t. all other items for the same application
+//Expected value ∈ [0-100]; preferably integral values
+//Monotonically increasing with larger values being considered better results

I would understand that apple is talking about integers when using the adjetive "integral" since the other math meaning makes no sense in this context. Would use a [return: BindAs (typeof (int?))]